### PR TITLE
staging-next 2023-09-07

### PR DESCRIPTION
**Pull request by vcunat**
##### Helpful links

- workflow docs: https://nixos.org/manual/nixpkgs/unstable/#submitting-changes-commit-policy
- important jobs:
  https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents
  https://hydra.nixos.org/job/nixos/staging-next-small/tested
- jobsets:
  https://hydra.nixos.org/jobset/nixpkgs/staging-next
  https://hydra.nixos.org/jobset/nixos/staging-next-small
- nix-review reports: https://malob.github.io/nix-review-tools-reports/
- previous staging-next: https://github.com/NixOS/nixpkgs/pull/252332#
- - - 
##### Mass breakages
(will be edited based on progress)

### CODE REVIEW

**Comment by fabianhjr** [comment]
There seems to be some breakage related to:

```
waf: error: no such option: --sbindir
```

**Comment by fabianhjr** [comment]
seems to come from multiple-outputs.sh which would be a stdenv rebuild UnU

```
pkgs/build-support/setup-hooks/multiple-outputs.sh
86-    prependToVar configureFlags \
87:        --bindir="${!outputBin}"/bin --sbindir="${!outputBin}"/sbin \
88-        --includedir="${!outputInclude}"/include --oldincludedir="${!outputInclude}"/include \
89-        --mandir="${!outputMan}"/share/man --infodir="${!outputInfo}"/share/info \
90-        --docdir="${!outputDoc}"/share/doc/"${shareDocName}" \
91-        --libdir="${!outputLib}"/lib --libexecdir="${!outputLib}"/libexec \
92-        --localedir="${!outputLib}"/share/locale
```

File: pkgs/development/tools/build-managers/waf/setup-hook.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ wafConfigurePhase() {
 
     local flagsArray=(
         $prefixFlag
-        $wafConfigureFlags "${wafConfigureFlagsArray[@]}"
+        $configureFlags "${configureFlagsArray[@]}"
```
Comment (fabianhjr) [inline pkgs/development/tools/build-managers/waf/setup-hook.sh:21]: here the sbindirs argument passes trought to waf, unsure about what to do to fix. :c

Comment (fabianhjr) [reply pkgs/development/tools/build-managers/waf/setup-hook.sh:21]: cc @AndersonTorres

**Comment by fabianhjr** [comment]
reverting 9f52275e52509dc667e6b42514ce1531416254a5 does allow the build to continue and isn't an stdenv rebuild. :S

File: pkgs/development/tools/build-managers/waf/setup-hook.sh (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/development/tools/build-managers/waf/setup-hook.sh:21]: It was reverted by https://github.com/NixOS/nixpkgs/pull/252874, among other things.

Comment (fabianhjr) [reply pkgs/development/tools/build-managers/waf/setup-hook.sh:21]: oh then that revert missed the staging-next cut, would be fine to revert here as well?

Comment (AndersonTorres) [reply pkgs/development/tools/build-managers/waf/setup-hook.sh:21]: A revert against other? I didn't understand.

Comment (fabianhjr) [reply pkgs/development/tools/build-managers/waf/setup-hook.sh:21]: I think what happened was:

1. first waf change was merged into staging
2. staging-next was cut from staging
3. second waf change reverting first waf was merged into staging-next

So the second change isn't on staging-next. :c 

So an option would be to merge this second change form staging into staging-next or reverting the first change from staging-next.

Comment (AndersonTorres) [reply pkgs/development/tools/build-managers/waf/setup-hook.sh:21]: The revert looks unnatural to me. I vote for the first.

Comment (yu-re-ka) [reply pkgs/development/tools/build-managers/waf/setup-hook.sh:21]: #254182

**Comment by AndersonTorres** [comment]
Oops! I broke it again!
There are two identical IDs in the docs, ironically called `similarVariables`, into `hooks/zig.section.md` and `hooks/waf.section.md`.
The fix is easy, but where should I insert it? `staging-next` maybe?

**Comment by AndersonTorres** [comment]
Hey, was this fixed? O.o

**Comment by AndersonTorres** [comment]
Again, solved on https://github.com/NixOS/nixpkgs/pull/254388

**Comment by vcunat** [comment]
But now we have a merge conflict in ` doc/hooks/waf.section.md`

**Comment by AndersonTorres** [comment]
Should I force-push my staging-next over that one?

**Comment by Artturin** [comment]
Pushed a 0 rebuild cross fix for iproute2

**Comment by vcunat** [comment]
Generally, please, only fast-forward pushes for `staging-next` and `staging`.  That's my intuition, as many people may be using those branches, e.g. as bases of pull requests.

**Review by yu-re-ka — APPROVED** [review]
The jobset looks very done

**Comment by vcunat** [comment]
Yes.  I see some regressions, but I'd say it's fairly good compared to average `staging-next` merges.  (I'm hurrying because of libwebp security.)

**Pull request merged**